eng/common/scripts/TypeSpec-Project-Sync.ps1

```diff
@@ -13,115 +15,133 @@ Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
 $sparseCheckoutFile = ".git/info/sparse-checkout"
 
 function AddSparseCheckoutPath([string]$subDirectory) {
-    if (!(Test-Path $sparseCheckoutFile) -or !((Get-Content $sparseCheckoutFile).Contains($subDirectory))) {
-        Write-Output $subDirectory >> .git/info/sparse-checkout
-    }
+  if (!(Test-Path $sparseCheckoutFile) -or !((Get-Content $sparseCheckoutFile).Contains($subDirectory))) {
+    Write-Output $subDirectory >> .git/info/sparse-checkout
+  }
 }
 
 function CopySpecToProjectIfNeeded([string]$specCloneRoot, [string]$mainSpecDir, [string]$dest, [string[]]$specAdditionalSubDirectories) {
-    $source = "$specCloneRoot/$mainSpecDir"
-    Copy-Item -Path $source -Destination $dest -Recurse -Force
-    Write-Host "Copying spec from $source to $dest"
+  $source = Join-Path $specCloneRoot $mainSpecDir
+  Copy-Item -Path $source -Destination $dest -Recurse -Force
+  Write-Host "Copying spec from $source to $dest"
 
-    foreach ($additionalDir in $specAdditionalSubDirectories) {
-        $source = "$specCloneRoot/$additionalDir"
-        Write-Host "Copying spec from $source to $dest"
-        Copy-Item -Path $source -Destination $dest -Recurse -Force
-    }
+  foreach ($additionalDir in $specAdditionalSubDirectories) {
+    $source = Join-Path $specCloneRoot $additionalDir
+    Write-Host "Copying spec from $source to $dest"
+    Copy-Item -Path $source -Destination $dest -Recurse -Force
+  }
 }
 
 function UpdateSparseCheckoutFile([string]$mainSpecDir, [string[]]$specAdditionalSubDirectories) {
-    AddSparseCheckoutPath $mainSpecDir
-    foreach ($subDir in $specAdditionalSubDirectories) {
-        AddSparseCheckoutPath $subDir
-    }
+  AddSparseCheckoutPath $mainSpecDir
+  foreach ($subDir in $specAdditionalSubDirectories) {
+    Write-Host "Adding $subDir to sparse checkout"
+    AddSparseCheckoutPath $subDir
+  }
 }
 
 function GetGitRemoteValue([string]$repo) {
-    Push-Location $ProjectDirectory
-    $result = ""
-    try {
-        $gitRemotes = (git remote -v)
-        foreach ($remote in $gitRemotes) {
-            if ($remote.StartsWith("origin")) {
-                if ($remote -match 'https://github.com/\S+') {
-                    $result = "https://github.com/$repo.git"
-                    break
-                } elseif ($remote -match "git@github.com:\S+"){
-                    $result = "git@github.com:$repo.git"
-                    break
-                } else {
-                    throw "Unknown git remote format found: $remote"
-                }
-            }
+  Push-Location $ProjectDirectory
+  $result = ""
+  try {
+    $gitRemotes = (git remote -v)
+    foreach ($remote in $gitRemotes) {
+      Write-Host "Checking remote $remote"
+      if ($remote.StartsWith("origin") -or $remote.StartsWith("main")) {
```

Comment:
What context do we need to support "main" as a remote?

Reply:
These are the remotes in the pipeline run of test repo which doesn't have `origin`. That's why I added main.
```
10:29:05.890 cmdout 	[generate.py] Checking remote integration	https://x-access-token:***@github.com/openapi-env-test/azure-sdk-for-java (fetch)
10:29:05.891 cmdout 	[generate.py] Checking remote integration	https://x-access-token:***@github.com/openapi-env-test/azure-sdk-for-java (push)
10:29:05.891 cmdout 	[generate.py] Checking remote main	https://x-access-token:***@github.com/openapi-env-test/azure-sdk-for-java (fetch)
10:29:05.891 cmdout 	[generate.py] Checking remote main	https://x-access-token:***@github.com/openapi-env-test/azure-sdk-for-java (push)
10:29:05.891 cmdout 	[generate.py] Checking remote secondary	https://x-access-token:***@github.com/openapi-env-test/azure-sdk-for-java (fetch)
10:29:05.892 cmdout 	[generate.py] Checking remote secondary	https://x-access-token:***@github.com/openapi-env-test/azure-sdk-for-java (push)
```

Reply:
OK the automation pipeline must be doing something unique when it does its clones as by default origin is always set to what is cloned.